### PR TITLE
added body to xml converter script

### PR DESCRIPTION
**Pull request by haseebeqx**
this one can act as an assistant in finding xxe.
it supports normal body, multipart body and json and convert them to xml
then it open manual request editor with this data.
further editing is required if it has to be useful for finding vulnerabilities  
it will also change the content type to application/xml

### CODE REVIEW

File: targeted/2xml4xxe.js [hunk 1]
```diff
@@ -0,0 +1,248 @@
+// A Script to convert Request body in normal form or JSON to XML
+// and set Content type to application/json.
```
Comment (thc202) [inline targeted/2xml4xxe.js]: Shouldn't be ```application/xml```?

File: targeted/2xml4xxe.js [hunk 2]
```diff
@@ -0,0 +1,248 @@
...
+// it uses manual request editor where you can edit the convered request
```
Comment (thc202) [inline targeted/2xml4xxe.js]: There's a typo (```convered```).

File: targeted/2xml4xxe.js [hunk 3]
```diff
@@ -0,0 +1,248 @@
...
+	else{
+		js = bodyToJson(reqb);
+		body += jsonToXML(js);
+	}
+	msg.setRequestBody(body);
+	header = msg.getRequestHeader();
+	header.setHeader(org.parosproxy.paros.network.HttpRequestHeader.CONTENT_TYPE,"application/xml");
+	header.setHeader(org.parosproxy.paros.network.HttpRequestHeader.CONTENT_LENGTH,body.length);
+	msg.setRequestHeader(header);
+	ext = new org.parosproxy.paros.extension.history.ExtensionHistory;
```
Comment (thc202) [inline targeted/2xml4xxe.js]: Use ```org.parosproxy.paros.control.Control.getSingleton().getExtensionLoader().getExtension(org.parosproxy.paros.extension.history.ExtensionHistory.NAME)``` instead, to obtain the existing ```ExtensionHistory```.

File: targeted/2xml4xxe.js [hunk 4]
```diff
@@ -0,0 +1,248 @@
...
+					
+			}
+				out += af
+				out = JSON.parse(out);
+			result = mergeJson(out,result);
+		}
+	}
+	else{
+		pairs = body.split('&');
+		pairs.forEach(function(i, pair){
```
Comment (thc202) [inline targeted/2xml4xxe.js]: The parameters are switched.

File: targeted/2xml4xxe.js [hunk 5]
```diff
@@ -0,0 +1,248 @@
...
+function toXml(key,value,att){ //pretify
+	if(att == null)
+		return ("<"+key+">"+value+"</"+key+">\n");
+	else{
+		if(value.slice(-1) == '\n')
+			value = value.substring(0,value.length-1);
+		return ("<"+key+">\n "+value+" \n</"+key+">\n");
+		}
+}
+function multiToJson(msg){
```
Comment (thc202) [inline targeted/2xml4xxe.js]: This method needs to be updated as per latest changes done in the other script.

File: targeted/2xml4xxe.js (hunk 3 above)
Comment (haseebeqx) [inline targeted/2xml4xxe.js]: Ok but all I want to do was manual request editor. I am at work. currently
all I have is a basic phone. I can not edit it :(

File: targeted/2xml4xxe.js (hunk 5 above)
Comment (haseebeqx) [inline targeted/2xml4xxe.js]: i dont get it

Comment (thc202) [inline targeted/2xml4xxe.js]: For example, the [```if```/```else``` statement at line 178](https://github.com/zaproxy/community-scripts/pull/18/files#diff-d0be30a3da16ed672d3de040641a79e6R178), the ```else``` block was removed in the other script (```json_csrf_poc_generator.js```).

Comment (haseebeqx) [inline targeted/2xml4xxe.js]: ok

**Pull request closed without merging**